libraries/ms-drivers/src/stm32f0xx/mcp3427_adc.c

```diff
@@ -0,0 +1,196 @@
+#include "mcp3427_adc.h"
+#include "fsm.h"
+#include "log.h"
+#include "mcp3427_adc_defs.h"
+#include "soft_timer.h"
+
+// Note: we always read from both channels before reporting, taking 100ms between reports. If timing
+// is an issue, an optimization for boards (i.e. solar) using only one channel is to only read that
+// channel, so reports take 50ms.
+
+#define MCP3427_FSM_NAME "MCP3427 FSM"
```

Comment:
still don't know why we bother with these, maybe we should just open a ticket to remove the fsm name but this is fine here